consensus/module_consensus_state_sync.go

```diff
@@ -22,20 +23,69 @@ func (m *consensusModule) GetNodeAddress() string {
 	return m.nodeAddress
 }
 
-// TODO(#352): Implement this function, currently a placeholder.
-// commitReceivedBlocks commits the blocks received from the blocksReceived channel
-// it is intended to be run as a background process
+// blockApplicationLoop commits the blocks received from the blocksResponsesReceived channel in state sync processes
 func (m *consensusModule) blockApplicationLoop() {
-	// runs as a background process in consensus module
-	// listens on the blocksReceived channel
-	// commits the received block
+	for blockResponse := range m.blocksResponsesReceived {
+		block := blockResponse.Block
+		maxPersistedHeight, err := m.maxPersistedBlockHeight()
+		if err != nil {
+			m.logger.Err(err).Msg("couldn't query max persisted height")
+			return
```

Comment:
If we really want `blockApplicationLoop` to be a backroudn process, shouldn't these be `continue` statements?